### PR TITLE
feat: add semver targeting support to local flag evaluation

### PR DESCRIPTION
**Pull request by dmarticus**
## Summary

- Add 9 semver comparison operators (`semver_eq`, `semver_neq`, `semver_gt`, `semver_gte`, `semver_lt`, `semver_lte`, `semver_tilde`, `semver_caret`, `semver_wildcard`) for local feature flag evaluation
- Implements the client-side evaluation to match PostHog/posthog#44596
- Parsing handles v-prefix, whitespace, pre-release suffixes, leading zeros, and 4-part versions

## Why not use a semver library?

Libraries like `packaging` and `semver` exist but aren't runtime dependencies of this SDK. The current runtime deps are intentionally minimal (`requests`, `six`, `python-dateutil`, `backoff`, `distro`, `typing-extensions`), and adding a dependency for ~20 lines of string splitting isn't worth the install footprint cost for SDK consumers. The Rust feature-flags service uses its ecosystem's `semver` crate, but Rust deps don't have the same user-facing cost as Python deps.

## Test plan

- [x] All 9 operators tested with positive/negative cases
- [x] Edge cases: v-prefix, whitespace, leading zeros, pre-release suffixes, partial versions, 4-part versions
- [x] Caret operator `^0.x.y` behavior per semver spec
- [x] Invalid input and missing property handling
- [x] All 115 tests pass (12 new + 103 existing)

### CODE REVIEW

File: posthog/feature_flags.py [hunk 1]
```diff
@@ -505,6 +505,79 @@ def compare(lhs, rhs, operator):
...
+        "semver_eq",
+        "semver_neq",
+        "semver_gt",
+        "semver_gte",
+        "semver_lt",
+        "semver_lte",
+        "semver_tilde",
+        "semver_caret",
+        "semver_wildcard",
+    ):
```
Comment (dmarticus) [inline posthog/feature_flags.py]: I know this is ugly, but the inline tuple approach matches the rest of the repo. I have refactored this in https://github.com/PostHog/posthog-python/pull/448 fwiw.

**Review by gustavohstrassburger — APPROVED** [review]
